cli/addmfadevice.go

```diff
@@ -0,0 +1,83 @@
+package cli
+
+import (
+	"fmt"
+
+	"github.com/99designs/keyring"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"gopkg.in/alecthomas/kingpin.v2"
+
+	"github.com/99designs/aws-vault/mfa"
+	"github.com/99designs/aws-vault/vault"
+)
+
+type AddMfaDeviceCommandInput struct {
+	ProfileName string
+	Keyring     keyring.Keyring
+	Config      vault.Config
+	Username    string
+}
+
+func ConfigureAddMfaCommand(app *kingpin.Application) {
+	input := AddMfaDeviceCommandInput{}
+
+	cmd := app.Command("add-mfa-device", "Adds a MFA device for the IAM user")
+	cmd.Arg("profile", "Name of the profile").
+		Required().
+		HintAction(awsConfigFile.ProfileNames).
+		StringVar(&input.ProfileName)
+
+	cmd.Arg("username", "Name of the user to add the MFA device for").
+		Required().
+		StringVar(&input.Username)
+
+	cmd.Action(func(c *kingpin.ParseContext) error {
+		input.Keyring = keyringImpl
+		input.Config.MfaTokenProvider = GlobalFlags.MfaTokenProvider
+		AddMFADeviceCommand(app, input)
+		return nil
+	})
+
+}
+
+func AddMFADeviceCommand(app *kingpin.Application, input AddMfaDeviceCommandInput) {
+	p, found := awsConfigFile.ProfileSection(input.ProfileName)
+	if !found {
+		app.Fatalf("Profile with name '%s' not found")
+	}
+
+	provider := &vault.KeyringProvider{
+		Keyring:         &vault.CredentialKeyring{Keyring: input.Keyring},
+		CredentialsName: p.Name,
+	}
+
+	masterCreds, err := provider.Retrieve()
+	if err != nil {
+		app.Fatalf("unable to retrieve master creds")
+	}
+
+	sess, err := session.NewSessionWithOptions(session.Options{
```

Comment:
You can use vault.NewSession for this
https://github.com/99designs/aws-vault/blob/f26b718f741a3f5c51809982539e6d7a1b6f4280/vault/vault.go#L20